### DISCUSSION TITLE
Harbor without Redis

### DISCUSSION THREAD

**Discussion by Vad1mo** [goharbor/harbor#15136] [Ideas]
How much effort would it take to move away from Redis and only use PGSQL?

At the moment, (to my knowledge), Harbor uses Redis for caching and data storage (Jobs).
Having to maintain Redis for such a small piece of functionality requires a significant maintenance overhead.

My suggestion is:

1. Remove the ephemeral requirement from Redis
2. Remove Redis from the Harbor Stack 

Fulfilling **1.** would be already a major step forward in reducing the complexity of the deployment.

Here are some reference that I could find for **1**, and **2** 
- https://github.com/golang/groupcache
- https://spin.atomicobject.com/2021/02/04/redis-postgresql/

**Comment by danielpacak** [comment]
Redis is also used by vulnerability scanners to (temporarily) cache scan results:
https://github.com/goharbor/harbor-scanner-clair
https://github.com/aquasecurity/harbor-scanner-trivy

**Comment by steven-zou** [comment]
Removing Redis means rewriting jobservice. That probably would be huge work. Meanwhile, Redis is used for session cache/registry metadata cache, removing Redis may influence the scalability of Harbor.

**Reply by Vad1mo** [reply]
Using Redis for caching is fine. But the fact that Redis is used to persisting data, while there is an existing DB where 99% of data is stored, makes no sense to me.

**Reply by steven-zou** [reply]
About persisting data in Redis, this issue we have discussed before. When we have an effort buffer, maybe the persist data can be removed from the Redis (and put into the psql).
